provider/caleta/model_mapper.go

```diff
@@ -209,11 +210,28 @@ func roundTransactionsMapper(roundTransactions *[]roundTransaction) *[]pam.Round
 			amt := toPamAmount(t.Payload.Amount)
 			txID := t.Payload.TransactionUUID
 			isGameOver := t.Payload.RoundClosed
+			refTx := t.Payload.ReferenceTransactionUUID
+
+			// Detect transaction type
+			var transactionType pam.TransactionType
+			switch {
+			case refTx == nil:
+				transactionType = pam.WITHDRAW
+			case t.Payload.Bet == "zero":
+				transactionType = pam.DEPOSIT
+			case t.Payload.Bet == "":
+				transactionType = pam.CANCEL
+			default:
+				log.Warn().Msgf("Failed to detect transactionType for txID=%s", txID)
```

Comment:
Until we have a better way of detecting this..

Reply:
👍